neo/Prompt/Commands/tests/test_send_command.py

```diff
@@ -38,129 +39,245 @@ def GetWallet1(cls, recreate=False):
 
-        self.assertEqual(2, len(res.Attributes))
+            self.assertTrue(framework)
+
+    def test_insufficient_funds(self):
 
-    def test_10_attributes(self):
         wallet = self.GetWallet1(recreate=True)
+        args = ['gas', self.watch_addr_str, '72620']
 
-        args = ['gas', self.watch_addr_str, '2', '--tx-attr=[{"usage":241,"data":"This is a remark"},{"usage":242,"data":"This is a remark 2"}]']
+        framework = construct_send_basic(wallet, args)
+        res = process_transaction(wallet, contract_tx=framework[0], scripthash_from=framework[1], fee=framework[2], owners=framework[3], user_tx_attributes=framework[4])
 
-        res = construct_and_send(None, wallet, args, prompt_password=False)
+        self.assertFalse(res)
 
-        self.assertTrue(res)
+    def test_bad_password(self):
+        with patch('neo.Prompt.Commands.Send.prompt', side_effect=['blah']):
 
-        self.assertEqual(3, len(res.Attributes))
+            wallet = self.GetWallet1(recreate=True)
+            args = ['neo', self.watch_addr_str, '50']
 
-    def test_11_bad_attributes(self):
-        wallet = self.GetWallet1(recreate=True)
+            framework = construct_send_basic(wallet, args)
+            res = process_transaction(wallet, contract_tx=framework[0], scripthash_from=framework[1], fee=framework[2], owners=framework[3], user_tx_attributes=framework[4])
+
+            self.assertFalse(res)
+
+    @patch.object(Send, 'gather_signatures')
+    def test_owners(self, mock):
+        with patch('neo.Prompt.Commands.Send.prompt', side_effect=[UserWalletTestCase.wallet_1_pass()]):
+
+            wallet = self.GetWallet1(recreate=True)
+
+            args = ['gas', self.wallet_1_addr, '2', "--owners=['AXjaFSP23Jkbe6Pk9pPGT6NBDs1HVdqaXK','APRgMZHZubii29UXF9uFa6sohrsYupNAvx']"]
+
+            framework = construct_send_basic(wallet, args)
+            process_transaction(wallet, contract_tx=framework[0], scripthash_from=framework[1], fee=framework[2], owners=framework[3], user_tx_attributes=framework[4])
+
+            self.assertTrue(mock.called)
 
-        args = ['gas', self.watch_addr_str, '2', '--tx-attr=[{"usa:241"data":his is a remark"}]']
+    def test_attributes(self):
+        with patch('neo.Prompt.Commands.Send.prompt', side_effect=[UserWalletTestCase.wallet_1_pass()]):
 
-        res = construct_and_send(None, wallet, args, prompt_password=False)
+            wallet = self.GetWallet1(recreate=True)
+            args = ['gas', self.watch_addr_str, '2', '--tx-attr={"usage":241,"data":"This is a remark"}']
 
-        self.assertTrue(res)
+            framework = construct_send_basic(wallet, args)
+            res = process_transaction(wallet, contract_tx=framework[0], scripthash_from=framework[1], fee=framework[2], owners=framework[3], user_tx_attributes=framework[4])
 
-        self.assertEqual(1, len(res.Attributes))
+            self.assertTrue(res)
+            self.assertEqual(2, len(res.Attributes))
```

Comment:
can you please add a comment above saying:
```
By default the script_hash of the transaction sender is added to the TransactionAttribute list, therefore the Attributes length is `count` + 1
```
We only have to do this once, but it's something that raises eyebrows if you read this code without it. Because it looks like you add just 1 attribute and then test for 2